column_computation.hh

```diff
@@ -36,7 +36,7 @@ public:
     virtual column_computation_ptr clone() const = 0;
 
     virtual bytes serialize() const = 0;
-    virtual bytes_opt compute_value(const schema& schema, const partition_key& key, const clustering_row& row) const = 0;
+    virtual bytes compute_value(const schema& schema, const partition_key& key) const = 0;
```

Comment:
Isn't the fact that we don't need a row just an artifact of the fact that we only support computed indexes on tokens of the partition key? If we add `CREATE INDEX ON users(lowercase(name))` then we suddenly do need the row.

@psarna

Reply:
Similarly, a general computation can return NULL. The token of a partition key can never be NULL, but that just happens to be the single computation we support.